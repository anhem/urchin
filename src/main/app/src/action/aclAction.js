import {Actions} from '../constants'
import {get, post} from './restClient'
import {notifyBackendError, notifySuccess} from "./notificationAction";

const {ACL} = Actions;

export const getAcl = (folderId) => (dispatch) => {
    dispatch({
        type: ACL.GET_ACL
    });
    return get('/api/permissions/acl/' + folderId)
        .then(json => dispatch({
            type: ACL.GET_ACL_SUCCESS,
            data: json
        }))
};

export const updateAclForGroup = (groupAcl) => (dispatch) => {
    dispatch({
        type: ACL.UPDATE_GROUP_ACL
    });
    return post('/api/permissions/acl/group', groupAcl)
        .then(json => {
                dispatch({
                    type: ACL.UPDATE_GROUP_ACL_SUCCESS,
                    data: json
                });
                notifySuccess("Success", "Group permissions updated");
            return dispatch(getAcl(groupAcl.folderId))
            }, error => {
                notifyBackendError(error);
            }
        )
};

export const updateAclForUser = (userAcl) => (dispatch) => {
    dispatch({
        type: ACL.UPDATE_USER_ACL
    });
    return post('/api/permissions/acl/user', userAcl)
        .then(json => {
                dispatch({
                    type: ACL.UPDATE_USER_ACL_SUCCESS,
                    data: json
                });
                notifySuccess("Success", "User permissions updated");
            return dispatch(getAcl(userAcl.folderId));
            }, error => {
                notifyBackendError(error);
            }
        )
};
