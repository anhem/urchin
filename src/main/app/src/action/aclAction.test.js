import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store'
import fetchMock from 'fetch-mock'
import {getAcl, updateAclForGroup, updateAclForUser} from "./aclAction";

describe('AclAction', () => {

    const folderId = 1;
    const getAclResponse = {
        groups: [],
        users: []
    };

    let store;

    beforeEach(() => {
        store = configureMockStore([thunk])({});

        fetchMock.get('/api/permissions/acl/' + folderId, getAclResponse);
    });

    afterEach(() => {
        fetchMock.reset();
        fetchMock.restore();
    });

    it('getAcl is successful', () => {

        return store.dispatch(getAcl(folderId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('updateAclForGroup is successful', () => {

        const groupAcl = {
            folderId: folderId
        };

        fetchMock.postOnce('/api/permissions/acl/group', groupAcl);

        return store.dispatch(updateAclForGroup(groupAcl)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('updateAclForUser is successful', () => {

        const userAcl = {
            folderId: folderId,
        };

        fetchMock.postOnce('/api/permissions/acl/user', userAcl);

        return store.dispatch(updateAclForUser(userAcl)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });
});