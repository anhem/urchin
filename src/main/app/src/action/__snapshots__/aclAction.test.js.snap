// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AclAction getAcl is successful 1`] = `
Array [
  Object {
    "type": "GET_ACL",
  },
  Object {
    "data": Object {
      "groups": Array [],
      "users": Array [],
    },
    "type": "GET_ACL_SUCCESS",
  },
]
`;

exports[`AclAction updateAclForGroup is successful 1`] = `
Array [
  Object {
    "type": "UPDATE_GROUP_ACL",
  },
  Object {
    "data": Object {
      "folderId": 1,
    },
    "type": "UPDATE_GROUP_ACL_SUCCESS",
  },
  Object {
    "type": "GET_ACL",
  },
  Object {
    "data": Object {
      "groups": Array [],
      "users": Array [],
    },
    "type": "GET_ACL_SUCCESS",
  },
]
`;

exports[`AclAction updateAclForUser is successful 1`] = `
Array [
  Object {
    "type": "UPDATE_USER_ACL",
  },
  Object {
    "data": Object {
      "folderId": 1,
    },
    "type": "UPDATE_USER_ACL_SUCCESS",
  },
  Object {
    "type": "GET_ACL",
  },
  Object {
    "data": Object {
      "groups": Array [],
      "users": Array [],
    },
    "type": "GET_ACL_SUCCESS",
  },
]
`;
